### PR TITLE
chore: preset IDs for pages

### PR DESCRIPTION
**Pull request by mmikita95**
Adding a mechanic to keep the count of pages in component trees, and to add predefined keys & IDs for pages if case if they're not provided by user. Fixes a bug with losing CMC page focus upon app code update.

### CODE REVIEW

File: src/streamsync/core_ui.py [hunk 1]
```diff
@@ -100,6 +103,8 @@ def ingest(self, serialised_components: Dict[str, Any]) -> None:
             self.components.pop(component_id)
         for component_id, sc in serialised_components.items():
             component = Component(**sc)
+            if component.type == "page" and component_id not in self.components:
```
Comment (FabienArcellier) [inline src/streamsync/core_ui.py:106]: Did you think to reuse attach here ?

File: src/streamsync/ui_manager.py [hunk 2]
```diff
@@ -137,6 +136,13 @@ def _create_component(component_tree: ComponentTree,  component_type: str, **kwa
     raw_content: dict = kwargs.pop("content", {})
     content = {key: _prepare_value(value) for key, value in raw_content.items()}
 
+    if component_type == "page" and "id" not in kwargs:
```
Comment (FabienArcellier) [inline src/streamsync/ui_manager.py:139]: Add a comment block that describe why we set automatic id on Page and not on other component

**Review by FabienArcellier — APPROVED** [review]
I have two remarks

File: src/streamsync/core_ui.py (hunk 1 above)
Comment (mmikita95) [reply src/streamsync/core_ui.py:106]: This was what I've tried initially, but it leads to constant re-attachment of the whole tree, including root components. Paints console red, to put it shortly 🥲

**Comment by ramedina86** [comment]
@FabienArcellier @mmikita95 are we good to merge?

**Comment by mmikita95** [comment]
@ramedina86 yes, good to go.

**Pull request merged**